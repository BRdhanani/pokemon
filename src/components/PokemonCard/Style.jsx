import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    borderWidth: 1,
    borderStyle: 'dashed',
    borderColor: '#2E3156',
    borderRadius: 8,
    margin: 10,
  },
  image: {
    width: 150,
    height: 150,
  },
  name: {
    fontWeight: '600',
    fontSize: 20,
    color: '#2E3156',
    textAlign: 'center',
    marginTop: 10,
  },
  number: {
    fontWeight: '400',
    fontSize: 20,
    color: '#2E3156',
    textAlign: 'center',
    marginBottom: 20,
  },
});

export default styles;
