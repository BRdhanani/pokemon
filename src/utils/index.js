export const getPokemonColour = {
  normal: '#DDCBD0',
  fighting: '#FCC1B0',
  flying: '#B2D2E8',
  poison: '#CFB7ED',
  ground: '#F4D1A6',
  rock: '#C5AEA8',
  bug: '#C1E0C8',
  ghost: '#D7C2D7',
  steel: '#C2D4CE',
  fire: '#EDC2C4',
  water: '#CBD5ED',
  grass: '#C0D4C8',
  electric: '#E2E2A0',
  psychic: '#DDCOCF',
  ice: '#C7D7DF',
  dragon: '#CADCDF',
  dark: '#C6C5E3',
  fairy: '#E4C0CF',
  unknown: '#CODFDD',
  shadow: '#CACACA',
};
