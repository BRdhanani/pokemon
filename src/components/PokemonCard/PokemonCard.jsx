import React, {useMemo} from 'react';
import {View, TouchableOpacity, Image, Text} from 'react-native';
import styles from './Style';
import {getPokemonColour} from '../../utils';

function PokemonCard({pokemon, number, navigation, pokemons}) {
  const gradients = useMemo(
    () => pokemon.types.map(type => getPokemonColour[type.type.name]),
    [pokemon],
  );

  const handleNavigate = () => {
    navigation.navigate('Pokemon', {
      pokemon,
      from: 'card',
      gradients,
      pokemons,
      index: number,
    });
  };

  return (
    <TouchableOpacity onPress={handleNavigate}>
      <View style={{...styles.container, backgroundColor: gradients[0]}}>
        <Image
          source={{
            uri: `https://img.pokemondb.net/sprites/home/normal/1x/${pokemon.name}.png`,
          }}
          style={styles.image}
        />
        <Text style={styles.name} testID="name">
          {pokemon.name}
        </Text>
        <Text style={styles.number} testID="number">
          {pokemon.id.toString().padStart(3, '0')}
        </Text>
      </View>
    </TouchableOpacity>
  );
}

export default PokemonCard;
