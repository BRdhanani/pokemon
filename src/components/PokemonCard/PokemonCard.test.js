import 'react-native';
import PokemonCard from './PokemonCard';
import {render} from '@testing-library/react-native';
import {pokemon, pokemons} from '../../utils/mock';

const pokemonCard = render(
  <PokemonCard pokemon={pokemon} pokemons={pokemons} />,
);

test('render PokemonCard component', () => {
  const {getByTestId} = pokemonCard;
  expect(getByTestId('name').props.children).toMatch('bulbasaur');
  expect(getByTestId('number').props.children).toMatch('001');
});
