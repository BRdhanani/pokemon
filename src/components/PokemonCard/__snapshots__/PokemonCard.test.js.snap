// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#C0D4C8",
        "borderColor": "#2E3156",
        "borderRadius": 8,
        "borderStyle": "dashed",
        "borderWidth": 1,
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
        "margin": 10,
      }
    }
  >
    <Image
      source={
        {
          "uri": "https://img.pokemondb.net/sprites/home/normal/1x/bulbasaur.png",
        }
      }
      style={
        {
          "height": 150,
          "width": 150,
        }
      }
    />
    <Text
      style={
        {
          "color": "#2E3156",
          "fontSize": 20,
          "fontWeight": "600",
          "marginTop": 10,
          "textAlign": "center",
        }
      }
    >
      bulbasaur
    </Text>
    <Text
      style={
        {
          "color": "#2E3156",
          "fontSize": 20,
          "fontWeight": "400",
          "marginBottom": 20,
          "textAlign": "center",
        }
      }
    >
      001
    </Text>
  </View>
</View>
`;
